### PR TITLE
fix: header_waiter batch synchronize worker address

### PR DESCRIPTION
**Pull request by akichidis**
Resolves: https://github.com/MystenLabs/narwhal/issues/286

Fixed the referenced issue. Added first `header_waiter` test for the batch synchronize happy path only.

### CODE REVIEW

File: primary/src/tests/header_waiter_tests.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
+use crate::{
+    common::create_db_stores,
```
Comment (huitseeker) [inline primary/src/tests/header_waiter_tests.rs:2]: License check

File: primary/src/header_waiter.rs [hunk 2]
```diff
@@ -178,7 +182,7 @@ impl<PublicKey: VerifyingKey> HeaderWaiter<PublicKey> {
                             }
                             for (worker_id, digests) in requires_sync {
                                 let address = self.committee
-                                    .worker(&author, &worker_id)
+                                    .worker(&self.name, &worker_id)
```
Comment (huitseeker) [inline primary/src/header_waiter.rs:185]: 😍

File: primary/src/tests/header_waiter_tests.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                            panic!("Error occurred {err}");
+                        }
+                    }
+                }
+                //  sender closed
+                _ => panic!("Failed to receive network message"),
+            }
+        }
+    })
+}
```
Comment (huitseeker) [inline primary/src/tests/header_waiter_tests.rs:134]: Could you move the `worker_listener` you have in primary/src/block_synchronizer/tests/block_synchronizer_tests.rs to primary/src/tests/common.rs, and reuse it here and in the block synchronizer?

**Review by huitseeker — COMMENTED** [review]
Thanks a crapton for the sleuthing and the fix!!!

File: primary/src/tests/header_waiter_tests.rs (hunk 3 above)
Comment (huitseeker) [reply primary/src/tests/header_waiter_tests.rs]: Done in https://github.com/MystenLabs/narwhal/pull/297/commits/f16b900c3260ce2d537ddd92e37d82ca783248af

Comment (akichidis) [reply primary/src/tests/header_waiter_tests.rs]: Thanks a lot for https://github.com/MystenLabs/narwhal/commit/01acea86c6f5a15abacf549e2905da9cb8bf392f

**Comment by akichidis** [comment]
@huitseeker let me know if you are happy with the changes and approve so can merge

**Review by huitseeker — APPROVED** [review]
Thanks a lot for fixing the integration test too!

**Pull request merged**